### PR TITLE
Update changelog and version for 0.4 in master branch

### PR DESCRIPTION
**Pull request by BoxiLi**

### CODE REVIEW

File: doc/source/changelog.rst [hunk 1]
```diff
@@ -3,6 +3,39 @@ Changelog
 *********
 
 
+Version 0.4.0 (Nov 23, 2024)
++++++++++++++++++++++++++++++
+
+This release adds a new quantum circuit renderer based on matplotlib and text and improves the circuit simulation speed.
```
Comment (hodgestar) [inline doc/source/changelog.rst:9]:
```suggestion
This release adds two new quantum circuit renderers -- one based on matplotlib and another that renders the circuit as text -- and improves the circuit simulation speed.
```
The "and text" felt a bit difficult to parse when reading.

File: doc/source/changelog.rst [hunk 2]
```diff
@@ -3,6 +3,39 @@ Changelog
...
+
+New features
+------------
+- Add matplotlib-based and text-based quantum circuit renderer. The new renderer is more flexible and can be used in different environments without latex dependencies. (`#239 <https://github.com/qutip/qutip-qip/pull/239>`_ and `#244 <https://github.com/qutip/qutip-qip/pull/244>`_ by Rushiraj Gadhvi)
```
Comment (hodgestar) [inline doc/source/changelog.rst:13]:
```suggestion
- Add matplotlib-based and text-based quantum circuit renderers. The new rendering framework is more flexible and can be used in different environments without latex dependencies. (`#239 <https://github.com/qutip/qutip-qip/pull/239>`_ and `#244 <https://github.com/qutip/qutip-qip/pull/244>`_ by Rushiraj Gadhvi)
```

File: doc/source/changelog.rst [hunk 3]
```diff
@@ -3,6 +3,39 @@ Changelog
...
+
+Improvements
+------------
+- `einsum` is used to speed up the circuit simulation. (`#225 <https://github.com/qutip/qutip-qip/pull/225>`_ by Boxi Li)
```
Comment (hodgestar) [inline doc/source/changelog.rst:17]: Do you have a sense of how much the speed up was? Might be nice to say "speed up the circuit simulation by up to 2x" or similar to give the reader a sense of the improvement. Also happy to leave as is if it's too difficult to give a number.

**Review by hodgestar — COMMENTED** [review]
Looking forward to the release. This is a big one.

File: doc/source/changelog.rst [hunk 4]
```diff
@@ -3,6 +3,39 @@ Changelog
...
+
+This release adds two new quantum circuit renderers -- one based on matplotlib and another that renders the circuit as text -- and improves the circuit simulation speed.
+
+New features
+------------
+- Add matplotlib-based and text-based quantum circuit renderers. The new rendering framework is more flexible and can be used in different environments without latex dependencies. (`#239 <https://github.com/qutip/qutip-qip/pull/239>`_ and `#244 <https://github.com/qutip/qutip-qip/pull/244>`_ by Rushiraj Gadhvi)
+
+Improvements
+------------
+- `einsum` is used to speed up the circuit simulation. (`#225 <https://github.com/qutip/qutip-qip/pull/225>`_ by Boxi Li)
```
Comment (BoxiLi) [inline doc/source/changelog.rst:17]:
```suggestion
- `einsum` is used to speed up the circuit simulation by up to 10 times. (`#225 <https://github.com/qutip/qutip-qip/pull/225>`_ by Boxi Li)
```

**Pull request merged**